llmfoundry/models/mpt/configuration_mpt.py

```diff
@@ -145,6 +177,35 @@ def __init__(
             init_config_defaults,
         )
 
+        if 'reuse_kv_layer_idx' in self.attn_config and self.attn_config[
+            'attn_impl'] == 'torch':
+            raise NotImplementedError(
+                'reusing kv cache from a previous layer is not implemented for torch attention.',
+            )
+        if block_overrides is not None:
+            warnings.warn(
```

Comment:
You can use https://github.com/mosaicml/llm-foundry/blob/21c9e0a58a68b2ddb7cc7bea40a11c9add85d2c9/llmfoundry/utils/warnings.py#L41 as the warning class

Reply:
Done.